src/service/enrichment/mod.rs

```diff
@@ -171,3 +174,77 @@ fn get_data(
 
     resp
 }
+
+// Global state for caching
+// static METADATA_CACHE: Lazy<Arc<RwLock<HashMap<String, EnrichmentTableMetadata>>>> =
+//     Lazy::new(|| Arc::new(RwLock::new(HashMap::new())));
+
+/// Retrieve enrichment table data
+pub async fn get_enrichment_table(
+    org_id: &str,
+    table_name: &str,
+) -> Result<Vec<vrl::value::Value>, anyhow::Error> {
+    let records = get_enrichment_table_json(org_id, table_name).await?;
+
+    Ok(records
+        .iter()
+        .map(crate::service::db::enrichment_table::convert_to_vrl)
+        .collect())
+}
+
+pub async fn get_enrichment_table_json(
+    org_id: &str,
```

Comment:
we don't need this, we still can use the old logic, call the search, it will be fine. you need to known, there is a super cluster, you can simple write like this.